pkg/users/local.js

```diff
@@ -37,18 +38,78 @@
+        handle.watch(() => {
+            getLogins().then(setDetails);
+        });
+        return handle.close;
+    }, [path, accounts, shadow]);
+
+    if (!accounts || !groups || !current_user_info || !details)
         return null;
 
-    if (path.length === 0)
-        return <AccountsList accounts={accounts} current_user={current_user_info.name} />;
-    else
+    // lastlog uses same sorting as /etc/passwd therefore arrays can be combined based on index
+    const accountsInfo = accounts.map((account, i) => {
+        return Object.assign({}, account, details[i]);
+    });
+
+    if (path.length === 0) {
+        return <AccountsMain accountsInfo={accountsInfo} current_user={current_user_info.name} groups={groups} />;
+    } else
         return (
-            <AccountDetails accounts={accounts} groups={groups} shadow={shadow}
+            <AccountDetails accounts={accountsInfo} groups={groups} shadow={shadow}
                             current_user={current_user_info.name} user={path[0]} />
         );
 }
 
+function get_locked(name) {
+    return cockpit.spawn(["/usr/bin/passwd", "-S", name], { environ: ["LC_ALL=C"], superuser: "require" })
+            .catch(() => "")
+            .then(content => {
+                const status = content.split(" ")[1];
+                // libuser uses "LK", shadow-utils use "L".
+                return status && (status == "LK" || status == "L");
+            });
+}
+
+async function getLogins() {
+    const lastlog = await cockpit.spawn(["/usr/bin/lastlog"], { environ: ["LC_ALL=C"] })
+            .catch(err => console.warn("Unexpected error when getting last login information", err));
+    const w = await cockpit.spawn(["/usr/bin/w", "-sh"], { environ: ["LC_ALL=C"] })
+            .catch(err => console.warn("Unexpected error when getting logged in accounts", err));
+
+    const currentLogins = w.split('\n').slice(0, -1).map(line => {
+        return line.split(/ +/)[0];
+    });
+
+    // drop header and last empty line with slice
+    const promises = lastlog.split('\n').slice(1, -1).map(async line => {
+        const splitLine = line.split(/ +/);
+        const name = splitLine[0];
+        const isLocked = await get_locked(name);
+
+        if (line.indexOf('**Never logged in**') > -1) {
+            return new Promise(resolve => {
+                resolve({ name: name, loggedIn: false, lastLogin: null, isLocked: isLocked });
+            });
+        }
+
+        const date_fields = splitLine.slice(-5);
+        return cockpit.spawn(["date", "+%s", "-d", date_fields.join(' ')], { environ: ["LC_ALL=C"], err: "out" })
+                .then(out => {
+                    return { name: name, loggedIn: currentLogins.includes(name), lastLogin: parseInt(out) * 1000, isLocked: isLocked };
+                })
+                .catch(e => console.warn(`Failed to parse date from lastlog line '${line}': ${e.toString()}`));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17830-20221124-161134-7e670612-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)